.github/workflows/rust.yml

```diff
@@ -96,7 +96,30 @@ jobs:
       - name: Check clippy
         working-directory: ${{ matrix.directory }}
         if: always()
-        run: cargo +${{ matrix.toolchain }} make clippy -- -D warnings
+        run: |
+          cargo +${{ matrix.toolchain }} make clippy --message-format=json \
+            | clippy-sarif \
+            | jq '.runs[].results |= unique' \
+            | tee clippy.sarif \
+            | sarif-fmt
+
+          jq -e '.runs[].results == []' clippy.sarif> /dev/null
+
+      - name: Process SARIF file
+        working-directory: ${{ matrix.directory }}
+        if: always()
+        run: |
+          cat clippy.sarif \
+            | jq --arg pwd "${{ matrix.directory }}" '.runs[].results[].locations[].physicalLocation.artifactLocation.uri |= $pwd + "/" + .' \
+            > clippy.sarif.tmp
+          mv clippy.sarif.tmp clippy.sarif
```

Comment:
We're running clippy in `matrix.directory`, so the files are relative to that location. By pretending this path to that location the SARIF file will be relative to the repository root.